src/backend/distributed/operations/shard_cleaner.c

```diff
@@ -75,6 +75,7 @@ TryDropMarkedShards(bool waitForCleanupLock)
 		/* rethrow as WARNING */
 		edata->elevel = WARNING;
 		ThrowErrorData(edata);
+		return -1;
```

Comment:
why do we return here and return `-1`?

Reply:
With the current structure, we don't need to return here I think because if we error, the return value from this function will be zero and we only check if the return value is positive in maintanence part:

```
			if (numberOfDroppedShards > 0)
			{
				ereport(LOG, (errmsg("maintenance daemon dropped %d distributed "
									 "shards previously marked to be removed",
									 numberOfDroppedShards)));
			}
```

Reply:
I meant with the original PR there were some other changes and `-1` was a leftover from there, now we don't this need anymore